src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -15252,40 +15252,26 @@ namespace SurfaceGeometry {
 
         constexpr Real64 TurnThreshold(0.000001); // Sensitivity of convexity test, in radians
 
-        int n;                                  // Loop index
-        int Np1;                                // Loop index
-        int Np2;                                // Loop index
-        Real64 Det;                             // Determinant for picking projection plane
-        Real64 DotProd;                         // Dot product for determining angle
-        Real64 Theta;                           // Angle between edge vectors
-        Real64 LastTheta;                       // Angle between edge vectors
-        Real64 V1len;                           // Edge vector length
-        Real64 V2len;                           // Edge vector length
+        Real64 LastTheta = 0.0;                 // Angle between edge vectors
         bool SignFlag;                          // Direction of edge turn : true is right, false is left
         bool PrevSignFlag(false);               // Container for the sign of the previous iteration's edge turn
+        bool PrevSignFlagInitialized(false);    // Whether we picked a PrevSignFlag already or not
```

Comment:
This is the minimal fix, see it in isolation in commit https://github.com/NREL/EnergyPlus/pull/9544/commits/f0d9a98e306cb5b50ccb260d38f70ad7f6da9c3b

the PrevSignFlag was set on the first iteration even if that first iteration happened to be a case of collinearity, meaning that PrevSignFlag was just set to the initial value of SignFlag = false, when no SignFlag could actually be computed!